pkg/localclusterimport/import_local_cluster.go

```diff
@@ -0,0 +1,240 @@
+	// simply obtaining the secret above will perform one layer of decoding.
+	// we need to manually perform another to ensure that the data is correctly copied to the new secret.
+	hubPullSecret.Data[".dockerconfigjson"], err = base64.StdEncoding.DecodeString(string(pullSecret.Data[".dockerconfigjson"]))
+	if err != nil {
+		i.log.Errorf("hubClusterDay2Importer: Unable to decode base64 pull secret data due to error %s", err.Error())
+		return err
+	}
+	hubPullSecret.OwnerReferences = []metav1.OwnerReference{}
+	_, err = i.clusterImportOperations.CreateSecret(LocalClusterNamespace, &hubPullSecret)
+	if err != nil {
+		i.log.Errorf("hubClusterDay2Importer: Unable to store hub pull secret due to error %s", err.Error())
+		return err
+	}
+
+	//Create an AgentClusterInstall in the local cluster namespace
+	userManagedNetworkingActive := true
+	agentClusterInstall := &hiveext.AgentClusterInstall{
+		Spec: hiveext.AgentClusterInstallSpec{
+			Networking: hiveext.Networking{
+				UserManagedNetworking: &userManagedNetworkingActive,
+			},
+			ClusterDeploymentRef: v1.LocalObjectReference{
+				Name: LocalClusterNamespace + "-cluster-deployment",
+			},
+			ImageSetRef: &hivev1.ClusterImageSetReference{
+				Name: "local-cluster-image-set",
+			},
+			ProvisionRequirements: hiveext.ProvisionRequirements{
+				ControlPlaneAgents: numberOfControlPlaneNodes,
+			},
+		},
+	}
+	agentClusterInstall.Namespace = LocalClusterNamespace
+	agentClusterInstall.Name = LocalClusterNamespace + "-cluster-install"
+	err = i.clusterImportOperations.CreateAgentClusterInstall(agentClusterInstall)
+	if err != nil {
+		i.log.Errorf("hubClusterDay2Importer: Could not create AgentClusterInstall due to error %s", err.Error())
+		return err
+	}
+
+	// Fetch the recently stored AgentClusterInstall so that we can obtain the UID
+	aci, err := i.clusterImportOperations.GetAgentClusterInstall(LocalClusterNamespace, LocalClusterNamespace+"-cluster-install")
+	if err != nil {
+		i.log.Errorf("hubClusterDay2Importer: Failed to fetch created AgentClusterInstall due to error %s", err.Error())
+		return err
+	}
+
+	// Create a cluster deployment in the local cluster namespace
+	clusterDeployment := &hivev1.ClusterDeployment{
+		Spec: hivev1.ClusterDeploymentSpec{
+			Installed: true, // Let ZTP know to import this cluster
+			ClusterMetadata: &hivev1.ClusterMetadata{
+				ClusterID:                "",
+				InfraID:                  "",
+				AdminKubeconfigSecretRef: v1.LocalObjectReference{Name: kubeConfigSecret.Name},
+			},
+			ClusterInstallRef: &hivev1.ClusterInstallLocalReference{
+				Name:    LocalClusterNamespace + "-cluster-install",
+				Group:   "extensions.hive.openshift.io",
+				Kind:    "AgentClusterInstall",
+				Version: "v1beta1",
+			},
+			Platform: hivev1.Platform{
+				AgentBareMetal: &agent.BareMetalPlatform{
+					AgentSelector: metav1.LabelSelector{
+						MatchLabels: map[string]string{"infraenv": fmt.Sprintf("%s-day2-infraenv", LocalClusterNamespace)},
```

Comment:
you can remove this part, you don't need to decide for the user what will be the infra env. 
you can write something like local cluster or something generic